backend/test/services/BaseService.test.ts

```diff
@@ -0,0 +1,206 @@
+import { initialize, Server } from '@projecttacoma/node-fhir-server-core';
+import express from 'express';
+import supertest from 'supertest';
+import { serverConfig } from '../../src/config/serverConfig';
+import { uploadTransactionBundle } from '../../src/services/BaseService';
+import { cleanUpTestDatabase, setupTestDatabase } from '../utils';
+
+let server: Server;
+
+const INVALID_METHOD_REQ = {
+  resourceType: 'Bundle',
+  type: 'transaction',
+  entry: [
+    {
+      resource: {
+        resourceType: 'Measure',
+        id: 'test-measure',
+        library: ['Library/test-library']
+      },
+      request: {
+        method: 'GET',
+        url: 'Measure/test-measure'
+      }
+    }
+  ]
+};
+
+const INVALID_PUT_REQ = {
+  resourceType: 'Bundle',
+  type: 'transaction',
+  entry: [
+    {
+      resource: {
+        resourceType: 'Measure',
+        library: ['Library/test-library']
+      },
+      request: {
+        method: 'PUT',
+        url: 'Measure/test-measure'
+      }
+    }
+  ]
+};
+
+const INVALID_RESOURCE_REQ = {
+  resourceType: 'Bundle',
+  type: 'transaction',
+  entry: [
+    {
+      resource: {
+        resourceType: 'Invalid',
+        id: 'test-library'
+      },
+      request: {
+        method: 'POST',
+        url: 'Library/test-library'
+      }
+    }
+  ]
+};
+
+const VALID_PUT_REQ = {
+  resourceType: 'Bundle',
+  type: 'transaction',
+  entry: [
+    {
+      resource: {
+        resourceType: 'Measure',
+        id: 'test-measure',
+        library: ['Library/test-library']
+      },
+      request: {
+        method: 'PUT',
+        url: 'Measure/test-measure'
+      }
+    }
+  ]
+};
+
+const VALID_POST_REQ = {
+  resourceType: 'Bundle',
+  type: 'transaction',
+  entry: [
+    {
+      resource: {
+        resourceType: 'Library',
+        id: 'test-library'
+      },
+      request: {
+        method: 'POST',
+        url: 'Library/test-library'
+      }
+    }
+  ]
+};
+
+describe('BaseService', () => {
+  beforeAll(async () => {
+    const app = express();
+    app.use(express.json({ limit: '50mb', type: 'application/json+fhir' }));
+    app.use(express.json({ limit: '50mb', type: 'application/fhir+json' }));
+    app.post('/:base_version/', (req, res, next) => {
+      return uploadTransactionBundle(req, res)
+        .then(result => res.status(200).json(result))
+        .catch(err => next(err));
+    });
+    server = initialize(serverConfig, app);
```

Comment:
I feel like there should be a way to avoid duplicating all this from `index.ts`. Maybe we can create a function in `index.ts` containing this code (kind of like we do in the bulk-export-server's [app.js](https://github.com/projecttacoma/bulk-export-server/blob/main/src/server/app.js)) and export it so that we can re-use it here

Reply:
Good suggestion! In order to do this, I had to pull this code out of `index.ts` and into its own file `app.ts`. Does that work?